pkg/codegen/nodejs/gen_program_test.go

```diff
@@ -1,11 +1,71 @@
 package nodejs
 
 import (
+	"os"
+	"path/filepath"
+	"strings"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+
 	"github.com/pulumi/pulumi/pkg/v3/codegen/internal/test"
+	"github.com/pulumi/pulumi/pkg/v3/testing/integration"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/executable"
 )
 
 func TestGenerateProgram(t *testing.T) {
-	test.TestProgramCodegen(t, "nodejs", GenerateProgram)
+	test.TestProgramCodegen(t, test.ProgramLangConfig{
+		Language:   "nodejs",
+		Extension:  "ts",
+		OutputFile: "index.ts",
+		Check:      nodejsCheck,
+		GenProgram: GenerateProgram,
+	})
+}
+
+func nodejsCheck(t *testing.T, path string) {
+	ex, err := executable.FindExecutable("yarn")
+	assert.NoError(t, err, "Could not find yarn executable")
+	dir := filepath.Dir(path)
+	name := filepath.Base(dir)
+	pkgName, pkgVersion := packagesFromTestName(name)
+	if pkgName == "" {
+		pkgName = "@pulumi/pulumi"
+		pkgVersion = "3.7.0"
+	}
+	pkg := pkgName + "@" + pkgVersion
+	defer func() {
+		nodeModules := filepath.Join(dir, "node_modules")
+		err = os.RemoveAll(nodeModules)
+		assert.NoError(t, err, "Failed to delete %s", nodeModules)
+		packageJSON := filepath.Join(dir, "package.json")
+		err = os.Remove(packageJSON)
+		assert.NoError(t, err, "Failed to delete %s", packageJSON)
+		yarnLock := filepath.Join(dir, "yarn.lock")
+		err = os.Remove(yarnLock)
+		assert.NoError(t, err, "Failed to delete %s", yarnLock)
+	}()
+	err = integration.RunCommand(t, "yarn add and install",
```

Comment:
Not seeing `yarn link @pulumi/pulumi` somewhere here, which would make it reference in-repo SDK.